### ISSUE TITLE
Joomla Update is STILL giving misleading information as of RC5

### ISSUE THREAD

**Issue by nikosdion** [joomla/joomla-cms#34968]
Attention @zero-24 @wilsonge 

I reported this issue for Joomla 4.0, you fixed it in 3.10 but it's still an issue in 4.0 as of RC5. Do remember that I explicitly told you that this must be fixed in Joomla 4 as well.

All my extensions have published update sites, the update sites claim compatibility with Joomla 4.0 and all extensions are up to date.

Still, extensions under “No Update Required” show one ‘Potential Upgrade Issue.’ per system, actionlog etc plugin included in the package. The More information text doesn't show as a tooltip but if it did show it reads “This extension includes a plugin that could cause the update to fail.<br><br>To perform the Joomla update safely you should either update this extension to a version compatible with your target version of Joomla or disable the relevant plugin(s) and check again.<br><br>For more information about the relevant plugins please check the 'Live Update' tab. ”

As I have already reported well over a month ago this information is not just wrong, it is misleading and defamatory. It is causing financial harm to third party developers. In short, it's breaking the law.

I see that you are no on a merge freeze and have a release date of August 17th. Let me put this in absolutely clear terms. If Joomla 4 releases on August 17th without having addressed this issue which automatically disseminates false and defamatory information against my software I am going to sue OpenSourceMatters Inc for defamation and loss of profit. I'd rather resolve this issue amicably in code than in court.

**Comment by richard67** [comment]
@nikosdion Was it this one which solved it for you? https://github.com/joomla/joomla-cms/pull/34776

**Comment by nikosdion** [comment]
Correct. That's the one that worked.

Please note that when transferring the code to Joomla 4 you will need to take care of the tooltip code being different in Joomla 3 (Bootstrap 2) and Joomla 4 (Bootstrap 5). When the code from the previous 3.10 alpha was copied to 4.0 RC4 this wasn't changed.

**Comment by richard67** [comment]
Thanks for confirming. There are much more differences meanwhile between the J3 and the J4 JS, so it's not a monkey patch.

**Comment by nikosdion** [comment]
Yes, I know. The J4 JS is plain vanilla whereas the J3 one is jQuery.

If it's not possible to make this change before J4 is released on August 17th then this feature needs to be completely removed, e.g. Joomla Update reverted to how it was on Joomla 4 RC3.

I am also worried about the overall direction of this component. It was meant to make Joomla updates easy and user friendly. The version I see in RC5 is anything but. The fact that the empty state and update confirmation page are the same almost caught _me_ off-guard. The fact that you have to confirm twice that you really do want to update, affirming things you can't reasonably affirm, is going to dissuade people from updating their sites. It's also a whole lot of theatre; as a developer I can claim my software supports Joomla 4.0, 4.1, etc so I don't have to retroactively change the update sites of two dozen extensions when 4.1 is eventually released. Whether an old version of my software will be compatible with a version of the CMS not yet released or, in fact, even written is anybody's guess. In the end of the day Joomla Update punishes diligent developers and screws over those of us who actually test our software before claiming compatibility. Note that NOBODY can test a new Joomla version before it's released since security patches are not public. So, basically, you are asking all 3PDs to be unscrupulous in what they claim to be supporting. The end result is worse than the previous state. Of course I don't expect you to get it; all of you maintainers are far removed from real world usage of the CMS and extensions development, actively avoid seeking input from your users and 3PDs and proceed to make changes without thinking second and third order effects.

**Comment by richard67** [comment]
> If it's not possible to make this change before J4 is released on August 17th

I think it should be possible.

> Of course I don't expect you to get it; all of you maintainers are far removed from real world usage of the CMS and extensions development, actively avoid seeking input from your users and 3PDs and proceed to make changes without thinking second and third order effects.

I will not further comment this insult.

**Comment by nikosdion** [comment]
I am sorry that you feel insulted when I am stating obvious facts. 

Maybe the facts are obvious to me because I am in direct connection with how your software is used in the real world as I am in direct, daily communication with end users, site builders and extension developers who use it. So let me tell you what is obvious to me about each cohort's interaction with Joomla Update's new features.

**Regular end users**. They are told they need to update their site immediately because of security. They go to their site and click on the Joomla Update link in the control panel page. Now they see a bureaucratic page filled with inscrutable information and warnings. Many will just think “okay, I can't update”. Those who try to plough through these inscrutable pages will only see a reference to extensions. They are then asked to confirm that all of their extensions are compatible with the next version. They will either balk out or inundate 3PDs with questions about compatibility since the design of the page and the message at the bottom are at odds with each other. All extensions up-to-date BUT I have to confirm they are compatible, therefore Joomla somehow knows that my extensions are not compatible with the new version. Even if they affirm something they cannot possibly vouch for, the update is not installed. Instead, yhey are met with a page that looks _exactly like_ the empty state page, misleading them into believing the update is installed. Those with a keen eye will see that they need to confirm they have taken a backup and their extensions are compatible (something they have already confirmed). So they will assume their extensions are ACTUALLY incompatible and their previous affirmation was ignored. See previous problem with 3PDs being inundated with requests. Some more people will now ask me why _my software_ displays this message instead of just taking a backup and why is it persisting after disabling the Backup on Update plugin or uninstalling my software to which I have to explain that it's Joomla showing that because Joomla tried its best to emulate Soviet bureaucracy. Very few people will make it to the update. 

Alternatively, people utterly confused by this HORRIBLE experience will seek help. Eventually, the Joomla Forum and 3PDs will tell them that Joomla Update is broken, its warnings are mostly invalid, they should just tick the two boxes and blindly install an update, thus nullifying this feature. The only end result is that people will correctly perceive Joomla updates as being exceedingly technical and confusing and are unlikely to keep on using Joomla. Remember that, in stark contrast to Joomla, WordPress updates automatically between point releases and with a single click on major versions. So why would someone use the super complicated and confusing Joomla when there's something more popular and far easier to use?

**Site integrators**. Had you talked to actual people building real world sites you'd know they can't and don't use off-the-shelf templates. They build their own or _customise_ an off-the-shelf one, giving it a different name. Since these templates do NOT have a public update source they will always appear as incompatible whenever their clients try to update their sites. This results in angry phone calls from the end client who perceives that the site builder sold them a site that needs to be rebuilt weeks to months after it first launched. From the site builder's perspective they're better off building in WordPress next time which doesn't lead to this breakdown in trust with their paying clients. Very few, if any, of the site builders will know how to create a fake update site for their template and make it lie that the template is compatible with major versions 4, 5, 6 through 10 of Joomla just to avoid this stupid problem that's losing them paying clients. Even the more technical developers who include custom modules and minor plugins won't know about this.

**Extension developers**. Those of us who are scrupulous will not claim compatibility of our extensions with a new Joomla minor version until we can actually test our software with it. As experience has shown, testing the RCs a couple of days before the actual release is a fool's errand; any b/c breaks come from security patches which are not available until the day of release. So on the day of release we have to simultaneously field questions from Joomla users asking us why our software appears as incompatible AND test with the released version which is a major undertaking. In the end of the day we have angry clients who can't update their sites. For those of us publishing security extensions having our software delay applying security releases because Joomla's security fixes may be breaking something and we need to test it is a surefire way to lose clients. So what we will have to do is to emulate the previous state: let people update Joomla regardless of our software so we don't have to field questions. This means that our software needs to LIE about which versions of Joomla it is supporting, completely nullifying the compatibility check feature.

So, to sum it up:
* Extension developers have to lie in their update sites to prevent losing clients, nullifying your feature.
* Site integrators now experience that the customisation they have done for a client's site appears as incompatible with any future version and are more likely to not use Joomla.
* Regular users are confused by the convoluted interface and EITHER avoid installing updates (therefore getting hacked, therefore abandoning Joomla as “insecure”) OR completely ignore the warnings, begrudging the bureaucratic interface and nullifying your feature. Or, of course, abandon Joomla for WordPress where updates are so much easier.

Are you telling me that you were aware of these issues having talked to site builders and extension developers but decided to include it regardless of the consequences? I find it hard to believe.

Remember that in all my admonitions to the Joomla project since 2009 I have only been wrong just ONCE about customising the backend menus and I was only wrong about the extent of the problem, only because most people still don't know this feature even exists (I have to thank the horrible UX and lack of marketing which made this feature practically invisible). Those who do did use this feature did get confused as I predicted for the reasons I predicted and did come to us 3PDs for support thinking our software is to blame as I predicted. I am neither a magician nor a clairvoyant. I just talk to the people using Joomla in the real world every day and can put myself in their shoes. This is called “being a consultant” which was, indeed, my profession before I started writing extensions for a living.
